### ISSUE TITLE
[Bug]: `generate_norm_flow_photons` returns wrong-length array when a module subset has zero detected photons

### ISSUE THREAD

**Issue by 472-Z** [Harvard-Neutrino/prometheus#111]
### Contact details

gloriaz@icecube.wisc.edu

### Location

`prometheus/photon_propagation/olympus/event_generation/photon_propagation/norm_flow_photons.py`, inside `make_generate_norm_flow_photons` -> `generate_norm_flow_photons`:

```python
if jnp.all(n_photons_masked == 0):
    times = [] * module_coords.shape[0]
    return ak.Array(times)
```

When every module passed into a given call has zero detected photons after the Poisson draw (`jnp.all(n_photons_masked == 0)`), the function returns `ak.Array([] * module_coords.shape[0])`. Because multiplying an empty Python list by an integer still yields an empty list (`[] * 200 == []`), the returned array has length **0**, regardless of `module_coords.shape[0]`.

This breaks the implicit contract that `generate_norm_flow_photons` always returns one entry per input module. When the detector is split into subsets for memory efficiency (e.g. in `generate_cascade` / `generate_realistic_track` in `event_generation.py`, or in downstream code that does its own module-batching around `pprop_func`), a subset that happens to register zero photons on every module contributes a length-0 array instead of its expected subset length. After the per-subset results are concatenated (`ak.concatenate`), the combined array ends up shorter than `det.module_coords.shape[0]`.

Any code downstream that consumes this result assuming a strict 1:1, index-aligned correspondence with `det.module_coords` / `detector.modules` (for example the module/string/OM index bookkeeping in `OlympusPhotonPropagator.propagate()`, which walks `res_event` and increments `string_idx`/`om_idx` by position) silently desyncs once the array comes back shorter than expected. No exception is raised; hits simply end up attributed to the wrong modules. In our own case this manifested as hits from a bright track appearing scattered uniformly across the whole detector instead of clustered near the track, which is what led us to trace it back to this array-length bug.

### Expected behavior

`generate_norm_flow_photons` should always return an `ak.Array` of length exactly `module_coords.shape[0]` (one entry, possibly empty, per input module), regardless of whether any photons were detected. The zero-photon branch should return:

```python
times = [[] for _ in range(module_coords.shape[0])]
```

so that the output length always matches the input module count and downstream index-based bookkeeping stays correctly aligned, even when concatenating results from multiple subsets.

### Steps to reproduce

1. Call `generate_norm_flow_photons(module_coords, module_efficiencies, source_pos, source_dir, source_time, source_nphotons, seed=...)` with a `module_coords` subset and a source configuration such that every module in that subset ends up with zero detected photons after the internal Poisson draw (e.g. all modules in the subset are far enough from the source that `ph_frac * source_photons * mod_eff_factor` rounds to ~0 for all of them, or restrict `module_coords` to a subset of distant/low-efficiency modules relative to the source).
2. Check `len(result)` — it will be `0` instead of `module_coords.shape[0]`.
3. To see the downstream corruption: split a larger detector into subsets via `generate_cascade` / `generate_realistic_track` (or equivalent custom batching) with a `splitter` value small enough that at least one subset is far from the source/track (e.g. `splitter` set below the total module count for a detector where the source is localized). Compare the resulting hit-to-module mapping against a single unsplit call (`splitter` set larger than the total module count) for the same event — the split version shows hits attributed to modules far from the track/source, while the unsplit version correctly clusters hits near the track/source.

### Relevant log output

```shell

```

### Side note (separate, smaller issue)

We also noticed `det.module_coords.shape[0] % splitter` (modulo) is used to compute the number of subsets in `generate_cascade`/`generate_realistic_track`, which doesn't look like the intended `ceil(det.module_coords.shape[0] / splitter)`. Flagging separately since it's a different line from the one above, though it's in the same splitting code path.

### Workaround

Setting `splitter` large enough that the split branch never triggers avoids the bug entirely, which is presumably why this hasn't surfaced more — the default `splitter=100000` in `generate_cascade`/`generate_realistic_track` means most realistic detector sizes never exercise the split path.

Happy to open a PR with the one-line fix if useful.

**Comment by MeighenBergerS** [comment]
This should be fixed with the recent PRs
